actions/ci-beholder-validator/action.yml

```diff
@@ -0,0 +1,101 @@
+name: "Schema Validator Action"
+description: "Validates schema compatibility with schema registry"
+
+inputs:
+  checkout-repo:
+    description: "Whether to checkout the repository"
+    required: false
+    default: "true"
+  checkout-repo-fetch-depth:
+    description: "Fetch depth for checkout"
+    required: false
+    default: "0"
+  docker-registry:
+    description: "Docker registry to use"
+    required: true
+    default: "aws"
+  # aws inputs
+  aws-role-duration-seconds:
+    description: ""
+    required: false
+    default: "900"
+  aws-region:
+    description: ""
+    required: false
+  aws-account-number:
+    description: ""
+    required: false
+  aws-role-arn:
+    description: ""
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      if: inputs.checkout-repo == 'true'
+      uses: actions/checkout@v4.2.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
+    - name: Detect changed files
+      id: changes
+      shell: bash
+      run:
+        echo "CHANGED_FILES=$(git diff --name-only origin/master)" >>
+        $GITHUB_ENV
```

Comment:
We typically use this action for changed files: https://github.com/dorny/paths-filter/

Here's a large example of usage: https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/ci-core.yml#L20-L110

Reply:
I went through the documentation and this seems to be extremely useful when you want changes with some filter condition. 
In my case, we just need to provide changed files and the actual filtering is done by the code. This is the relevant piece of code https://github.com/smartcontractkit/atlas/blob/7acd79f562164d05f7b1d389ba74b40248a574d7/beholder/schema_validator/cmd/validate.go#L62

Does it make sense to change this in that case? LMK what you think?

Reply:
No that's fine, your solution is good.